src/modules/Nis.rb

```diff
@@ -1053,72 +1054,90 @@ def WritePluses
       true
     end
 
-    # Configures the name service switch for the user databases
-    # according to chosen settings
-    # @return success?
+    # Configures the name service switch for the user databases according to chosen settings
+    #
+    # @return [Boolean] true on success; false otherwise
     def WriteNssConf
       dbs = ["passwd", "group", "shadow"]
+
       nis_dbs = ["services", "netgroup", "aliases"]
+
       # Why bother with both compat and nis?
       # If there's no plus, we don't have to write passwd etc.
       # And it's supposed to be faster.
       # But then programs have to reread nsswitch :( #23203
       # so we stick with compat.
       if @start
         # we want to switch to "compat"
-        Builtins.foreach(dbs) do |db|
+        dbs.each do |db|
           # what if a db is not mentioned?
           # We get [] meaning compat, so it's ok to make it explicit
           db_l = Nsswitch.ReadDb(db)
-          if !Builtins.contains(db_l, "compat")
+
+          if db_l.include?("compat")
```

Comment:
Wow, almost 6 months later I have just noticed we translated...

`!Builtins.contains(db_l, "compat")`

...into...

 `db_l.include?("compat")`

Without the "!", so we inverted the logic here!